File: SistemaDecomisos/src/java/model/ActaDonacion.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

import java.io.Serializable;
import java.util.Date;


/**
 *
 * @author Mery Zúñiga
 */
public class ActaDonacion implements Serializable,Jsonable{
    private int idDonacion;
    private Date fecha;
    private String institucion;
    private Policia policia;
    private ActaDecomiso decomiso;
    private String detalles;

    public ActaDonacion(){
        
    }
    
    public ActaDonacion(int idDonacion, Date fecha, String institucion, Policia policia, ActaDecomiso decomiso, String detalles) {
        this.idDonacion = idDonacion;
        this.fecha = fecha;
        this.institucion = institucion;
        this.policia = policia;
        this.decomiso = decomiso;
        this.detalles = detalles;
    }

    public int getIdDonacion() {
        return idDonacion;
    }

    public Policia getPolicia() {
        return policia;
    }

    

    public String getInstitucion() {
        return institucion;
    }

    

    public ActaDecomiso getDecomiso() {
        return decomiso;
    }

    public void setIdDonacion(int idDonacion) {
        this.idDonacion = idDonacion;
    }

    public void setPolicia(Policia policia) {
        this.policia = policia;
    }

    

    public void setInstitucion(String institucion) {
        this.institucion = institucion;
    }


    public void setDecomiso(ActaDecomiso decomiso) {
        this.decomiso = decomiso;
    }
    
    public Date getFecha() {
        return fecha;
    }

    public void setFecha(Date fecha) {
        this.fecha = fecha;
    }

    public String getDetalles() {
        return detalles;
    }

    public void setDetalles(String detalles) {
        this.detalles = detalles;
    }
    
    
}

File: SistemaDecomisos/web/js/objects/ActaDonacion.js
function ActaDonacion(idDonacion, fecha, institucion, policia, decomiso, detalles) {
    this.ActaDonacion(idDonacion, fecha, institucion, policia, decomiso, detalles);
}

ActaDonacion.prototype = {
    idDonacion: 0, 
    fecha: "",
    institucion: "",
    policia: "",
    decomiso: "",
    detalles: "",
    ActaDonacion: function (idDonacion,fecha, institucion, policia, decomiso, detalles) {
        this.idDonacion = idDonacion;
        this.fecha = fecha;
        this.institucion = institucion;
        this.policia = policia;
        this.decomiso = decomiso;
        this.detalles = detalles;
    }
};


ActaDonacion.from = function (plain) {
    acta = new ActaDonacion(
    plain.idDonacion,
    plain.fecha,
    plain.institucion, 
    plain.policia,
    plain.decomiso,
    plain.detalles);
    return acta;
};

ActaDonacion.to = function (acta) {
    return {
        _class: 'ActaDonacion',
        idDonacion: acta.idDonacion, 
        fecha: acta.fecha,
        institucion: acta.institucion, 
        policia: acta.policia, 
        decomiso: acta.decomiso,
        detalles: acta.detalles
    };
};